lib/ciChecks.js

```diff
@@ -0,0 +1,60 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Check for CI test failure and pings PR author.
+ * @param {import('probot').Context} context
+ */
+const handleFailure = async (context) => {
+  /**
+   * @type {import('probot').Octokit.ChecksGetSuiteResponse} checkSuite
+   */
+  const checkSuite = context.payload.check_suite;
+  if (checkSuite.conclusion === 'failure') {
+    // A checksuite might fail due to a merge to develop. We don't need to
+    // check for these since they are not failures related to a pull request.
+    if (checkSuite.pull_requests.length > 0) {
+      // The response from the payload does not include the complete pull request
+      // data, hence we need to fetch the pull request so we can get the author.
+      const pullRequestData = checkSuite.pull_requests[0];
+      const pullRequestResponse = await context.github.pulls.get(
+        context.repo({
+          pull_number: pullRequestData.number,
+        })
+      );
+      const pullRequest = pullRequestResponse.data;
+      const prAuthor = pullRequest.user.login;
+      const commentParams = context.repo({
+        issue_number: pullRequest.number,
+        body:
+          'Hi @' +
+          prAuthor +
+          ', there are some failing CI checks in your latest push, PTAL' +
```

Comment:
Can we also add a link to the failing test?

Reply:
Also, add in the comment that if you think that the test is a flake please file an issue before restarting.

Reply:
> Also, add in the comment that if you think that the test is a flake please file an issue before restarting.

Done

> Can we also add a link to the failing test?

No, because what we are listening to is the check suite and not the individual check run.
I didn't think it will be okay for oppiabot to comment on every failing test which is why I'm checking for a failure in the check suite.

Reply:
Ok, makes sense.